Improve course instance access/enrollment docs (#11180)
PrairieLearn/PrairieLearn: docs/courseInstance.md

```json title="infoCourseInstance.json"
{
  "allowAccess": [
    {
      "startDate": "2015-01-19T00:00:01",
      "endDate": "2015-05-13T23:59:59",
      "institution": "Any"
    }
  ]
}
```

### Controlling access by UID

You can restrict access to particular students by listing their UIDs in the `uids` property. For instance, the following rule only allows `student1@example.com` and `student2@example.com` to access the course instance between the specified dates:

```json title="infoCourseInstance.json"
{
  "allowAccess": [
    {
      "startDate": "2015-01-19T00:00:01",
      "endDate": "2015-05-13T23:59:59",
      "uids": ["student1@example.com", "student2@example.com"]
    }
  ]
}
```

## Enrollment controls

Students can enroll in a course instance through one of two ways:

1. They can use a URL specific to the course instance or [to one of its assessments](assessment/index.md#linking-to-assessments). You can find the "student link" on the "Settings" tab of the course instance. This link points students to the list of assessments associated to the course instance, enrolling them automatically in the course instance if they are not yet enrolled.

> 2. They can use the "Add or remove courses" button on PrairieLearn's homepage. This button opens a page listing all course instances that are currently available for enrollment, giving students the option to add new courses.

Reviewer: There's also the following button, right? Should we change this one to also be "Add or remove courses" for consistency?
<img width="646" alt="image" src="https://github.com/user-attachments/assets/486629a8-7b83-44a7-9912-25c3a8b1119c" />

Author: Good catch, I agree we want this to be consistent. Done in 6a39d65 (didn't see the need to split this into a separate PR).

> Some instructors may wish to hide their course from the list of available course instances in the Enroll page. This may be done to provide a small level of control over which students get access to the course, or to avoid confusion in case of course instances that are not expected to be visible to students in general. For these instances, the following setting will hide the course instance from the list of instances in the Enroll page, even if the instance is available for enrollment.

Reviewer: In this paragraph we are referring to the page reached from the "Add or remove courses" button as the "Enroll" page. Should we try and standardize this terminology a bit more?

Author: I agree we should standardize, though I'm not sure what to call this page. The card on the page has the header "courses". How about this (avoid referring to it directly if we can, but call it the "add/remove courses page" otherwise)?

```suggestion
Some instructors may wish to hide their course from the list of available course instances. This may be done to provide a small level of control over which students get access to the course, or to avoid confusion in case of course instances that are not expected to be visible to students in general. For these instances, the following setting will hide the course instance from the list of instances on the add/remove courses page, even if the instance is available for enrollment.
```

Reviewer: lgtm!

}
```

> Note that _this is not a security setting_. Students may still enroll in the course instance if they get access to the URL, either through friends or by [forced browsing attacks](https://owasp.org/www-community/attacks/Forced_browsing). Instructors that wish to actually restrict course enrollment to a specific list of students should instead use well-defined access rules with restrictions by institution or UIDs, described under ["Access controls"](#access-controls).

Reviewer: I think the "forced browsing attacks" sounds scary and isn't really adding anything of value here. I also think we can link directly to the UID access control section, which is what I think is relevant here?
```suggestion
Note that _this is not a security setting_. Students may still enroll in the course instance if they get access to the URL, such as from a friend. Instructors that wish to actually restrict course enrollment to a specific list of students should instead use an access rules with an explicit list of UIDs, as described under ["Controlling access by UID"](#controlling-access-by-uid).
```
